.github/workflows/pull-request.yml

```
@@ -0,0 +1,265 @@
+        run: npm install
+      
+      - name: Get changed TypeScript files
+        id: changed-files
+        uses: tj-actions/changed-files@v40
+          
+      - name: Run tests
+        if: steps.changed-files.outputs.only_changed != 'true'
+        run: npm run test -- --watchAll=false --coverage       
+      
+      - name: TypeScript compilation for changed files
+        run: |
```

actionlint (1.7.3)

shellcheck reported issue in this script: SC2043:warning:1:13: This loop will only ever run once. Bad quoting or missing glob/expansion?

(shellcheck)